lib/Arch/X86/Runtime/CMakeLists.txt

```diff
@@ -35,14 +35,20 @@ function(add_runtime_helper target_name address_bit_size enable_avx enable_avx51
     set(required_cpp_standard "c++11")
   endif()
 
+  if (address_bit_size EQUAL 32)
```

Comment:
This was broken as it was using `x86_64` in the target triple even for 32-bit.

Reply:
I think the prior reason was two-fold:
1) on x86-64, we know that trivial structs in C++, e.g. what we use for the register, e.g. Rn<...> and RnW<...>, would turn into basic integral or pointer arguments in the bitcode. On a 32-bit target, this wasn't guaranteed.

2) We couldn't guarantee i386 headers were present. You merging and extension of msp430 stuff into the headers might have resolved that.

I can see this making a material difference now that there is inline assembly with register constraints. What I'm less certain of are the end-to-end impact on the lifters. But if the tests still pass then maybe it's all fine?

Reply:
We only cross-compile the hyper calls (everything else in the runtime just uses the host machine's target triple). Then we `llvm-link` everything together. I believe if I use `llvm-dis` on the resulting runtime, the target triple is that of the host machine, which was how it worked before my hypercall implementation. So I don't think this should have an effect, except fixing how the hyper calls are compiled.